### PR TITLE
Remove document.write() from loader and stability improvements 

### PR DESCRIPTION
**Pull request by MrGoumX**

### CODE REVIEW

File: src/runnable.coffee [hunk 1]
```diff
@@ -10,6 +10,8 @@ define ['settings'], (Settings) ->
     then discards it.
     ###
     @run: ->
+      return @ unless Settings.window[Settings.command_queue_name].q
```
Comment (itholoenos) [inline src/runnable.coffee:13]: You can keep the queue in a variable and replace all references in the `run` method

```suggestion
      queue = Settings.window[Settings.command_queue_name]?.q

      return @ unless queue
```

File: spec/analytics_spec.coffee [hunk 2]
```diff
@@ -32,37 +32,77 @@ describe 'Analytics', ->
...
+      afterEach ->
+        @session_run_stub.restore()
 
-    context 'when a Session is acquired', ->
-      it 'creates an ActionsManager using new', ->
-        expect(@ActionsManager).to.be.calledWithNew
+      it 'creates a PluginsManager using new', ->
+        expect(@PluginsManager).to.be.calledWithNew
+
+      it 'tries to acquire a Session', ->
```
Comment (itholoenos) [inline spec/analytics_spec.coffee:59]: The session is already acquired as implied from the context. Do you want to test if the `Session` object is initialized properly?

Comment (MrGoumX) [reply spec/analytics_spec.coffee:59]: I haven't changed this context. The bad context name is [this](https://github.com/skroutz/analytics.js/pull/82/files/91ee101025e500ac5f02b69644a2e42b28a38a94#diff-216799ae6dd5fb7a75ac24d1ccf6eacf1a338dcccf19db3c4c65a9a92426a9f3R44), I will change it to something that makes sense.

File: src/runnable.coffee (hunk 1 above)
Comment (mrwhizzy) [reply src/runnable.coffee:13]: +1 strongly agree for something like:

```coffee
@run: ->
  queue = Settings.window[Settings.command_queue_name].q
  return @ unless queue

  for _, cmd of queue.slice()
    [category, type, args...] = cmd

    if @_commands[category]?[type]?
      @_commands[category][type].apply(@, args)
      queue.splice(queue.indexOf(cmd), 1)

  return @
```

Comment (MrGoumX) [reply src/runnable.coffee]: Done!

File: src/analytics.coffee [hunk 3]
```diff
@@ -6,17 +6,41 @@ define [
...
+        @_handle_session()
+      else
+        ###
+        Replaces our queue appending global function with one which immediately
+        runs commands when the commands array is empty or undefined.
+        Intended for programmatic usage and/or lazy command declaration.
+        ###
+        Settings.window[Settings.command_queue_name] = =>
+          (Settings.window[Settings.command_queue_name].q ||= []).push(arguments)
+          @_handle_session()
```
Comment (mrwhizzy) [inline src/analytics.coffee:23]: wouldn't it be better to make this a `_replace_queue` function and call it in the two places we need it?

Comment (MrGoumX) [reply src/analytics.coffee:23]: I think the extraction here is unnecessary, we know the code's purpose. Despite that I tried to extract it and we would have to update a lot of specs outside of analytics_spec.coffee as well and that makes the change even more complex.

Comment (mrwhizzy) [reply src/analytics.coffee:23]: alright understood, we can keep it as it is then

**Review by mrwhizzy — APPROVED** [review]
+1 LGTM!

**Review by itholoenos — APPROVED** [review]
+1 LGTM!

File: src/loader.coffee [hunk 4]
```diff
@@ -18,43 +13,16 @@ library without blocking the `window.onload` event.
...
-
-  doc.open()._l = ->
-    js = this.createElement("script")
-
-    this.domain = dom if(dom)
-    js.id = "js-iframe-async"
-    js.src = "@@analytics_base_url/#{url}"
-    this.body.appendChild(js)
-
-  doc.write('<body onload="window.inDapIF=true; document._l();">')
```
Comment (harrisred) [inline src/loader.coffee:57]: We should also update the [intro.js](https://github.com/skroutz/analytics.js/blob/master/tasks/wrappers/analytics/intro.js#L4)

Comment (MrGoumX) [reply src/loader.coffee:57]: Done, but I'm not sure if we edit these files by hand.

Comment (harrisred) [reply src/loader.coffee:57]: Discussed offline. We will move forward with the change.

**Review by harrisred — APPROVED** [review]
LGTM 👏👏👏👏 🚀

**Pull request merged**